conda_build/post.py

```diff
@@ -1406,8 +1412,19 @@ def check_overlinking_impl(
                     list(diffs)[1:3],
                 )
                 sysroots_files[srs] = sysroot_files
+
+    def sysroot_matches_subdir(path):
```

Comment:
Let's include a comment here with an example of what the `sysroot` path will look like. Make it easier to understand what we are slicing out of the pathname.